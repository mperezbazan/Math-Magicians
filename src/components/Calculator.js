import React from 'react';
import Button from './Button';
import Result from './Result';

const initialButtons = [
  { value: 'AC' },
  { value: '±' },
  { value: '%' },
  { value: '÷', color: 'orange' },
  { value: '7' },
  { value: '8' },
  { value: '9' },
  { value: 'x', color: 'orange' },
  { value: '4' },
  { value: '5' },
  { value: '6' },
  { value: '-', color: 'orange' },
  { value: '1' },
  { value: '2' },
  { value: '3' },
  { value: '+', color: 'orange' },
  { value: '0', columns: 2 },
  { value: '.' },
  { value: '=', color: 'orange' },
];
class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = { buttons: initialButtons };
  }

  render() {
    const { buttons } = this.state;
    return (
      <div className="calculator-container">
        <Result value={0} />
        {buttons.map((button) => (
          <Button
            color={button.color}
            value={button.value}
            columns={button.columns}
            key={button.value}
          />
        ))}
      </div>
    );
  }
}

export default Calculator;
