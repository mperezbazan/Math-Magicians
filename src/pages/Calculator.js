import React from 'react';
import CalculatorContainer from '../components/CalculatorContainer';

const Calculator = () => (
  <CalculatorContainer />
);

export default Calculator;
