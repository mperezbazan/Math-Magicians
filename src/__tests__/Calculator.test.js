import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

test('Match Calculator Snapshot', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
