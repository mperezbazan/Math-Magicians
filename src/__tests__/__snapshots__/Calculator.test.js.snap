// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Calculator Snapshot 1`] = `
<div
  className="calculator-container"
>
  <div
    className="result dark-grey col-4"
  >
    0
  </div>
  <button
    className="button grey col-1"
    onClick={[Function]}
    type="button"
    value="AC"
  >
    AC
  </button>
  <button
    className="button grey col-1"
    onClick={[Function]}
    type="button"
    value="+/-"
  >
    +/-
  </button>
  <button
    className="button grey col-1"
    onClick={[Function]}
    type="button"
    value="%"
  >
    %
  </button>
  <button
    className="button orange col-1"
    onClick={[Function]}
    type="button"
    value="รท"
  >
    รท
  </button>
  <button
    className="button grey col-1"
    onClick={[Function]}
    type="button"
    value="7"
  >
    7
  </button>
  <button
    className="button grey col-1"
    onClick={[Function]}
    type="button"
    value="8"
  >
    8
  </button>
  <button
    className="button grey col-1"
    onClick={[Function]}
    type="button"
    value="9"
  >
    9
  </button>
  <button
    className="button orange col-1"
    onClick={[Function]}
    type="button"
    value="x"
  >
    x
  </button>
  <button
    className="button grey col-1"
    onClick={[Function]}
    type="button"
    value="4"
  >
    4
  </button>
  <button
    className="button grey col-1"
    onClick={[Function]}
    type="button"
    value="5"
  >
    5
  </button>
  <button
    className="button grey col-1"
    onClick={[Function]}
    type="button"
    value="6"
  >
    6
  </button>
  <button
    className="button orange col-1"
    onClick={[Function]}
    type="button"
    value="-"
  >
    -
  </button>
  <button
    className="button grey col-1"
    onClick={[Function]}
    type="button"
    value="1"
  >
    1
  </button>
  <button
    className="button grey col-1"
    onClick={[Function]}
    type="button"
    value="2"
  >
    2
  </button>
  <button
    className="button grey col-1"
    onClick={[Function]}
    type="button"
    value="3"
  >
    3
  </button>
  <button
    className="button orange col-1"
    onClick={[Function]}
    type="button"
    value="+"
  >
    +
  </button>
  <button
    className="button grey col-2"
    onClick={[Function]}
    type="button"
    value="0"
  >
    0
  </button>
  <button
    className="button grey col-1"
    onClick={[Function]}
    type="button"
    value="."
  >
    .
  </button>
  <button
    className="button orange col-1"
    onClick={[Function]}
    type="button"
    value="="
  >
    =
  </button>
</div>
`;
